lib/api/node.go

```diff
@@ -63,11 +63,19 @@ type NodeBGPSpec struct {
 
 	// IPv4Address is the IPv4 address of this node.  At least one of the
 	// IPv4 and IPv6 addresses should be specified.
-	IPv4Address *net.IP `json:"ipv4Address,omitempty" validate:"omitempty,ipv4"`
+	IPv4Address *net.IP `json:"ipv4Address,omitempty" validate:"omitempty"`
+
+	// IPv4Network is the IPv4 local network that this node is on (corresponding
+	// to the configured IPv4 address).
+	IPv4Network *net.IPNet `json:"ipv4Network,omitempty" validate:"omitempty"`
```

Comment:
So, options:

- (1) We leave it like it is in the PR.
- (2) We move the IPv4 / IPv6Network fields out of BGP info, but still duplicating information.
- (3) We convert IPv4Address into a `string` on the API and force clients to update (nobody really uses this yet though so this is sorta OK). 
  - Add validation to make sure it's a valid CIDR and/or IP address.  
  - Display missing subnet length as a /32 for compat with earlier versions.

(3) requires that your BGP address falls within the CIDR you've configured for ipip.  If we want to allow BGP addresses to differ from the IPIP subnet then this won't fly, but I don't think we do.

I think I'm leaning towards (3).  WDYT?